lib/DevTools.ts

```diff
@@ -0,0 +1,104 @@
+type DevtoolsOptions = {
+    maxAge?: number;
+    name?: string;
+    postTimelineUpdate?: () => void;
+    preAction?: () => void;
+    logTrace?: boolean;
+    remote?: boolean;
+};
+
+type DevtoolsSubscriber = (message: {type: string; payload: unknown; state: string}) => void;
+
+type DevtoolsConnection = {
+    send(data: Record<string, unknown>, state: Record<string, unknown>): void;
+    init(state: Record<string, unknown>): void;
+    unsubscribe(): void;
+    subscribe(cb: DevtoolsSubscriber): () => void;
+};
+
+const ERROR_LABEL = 'Onyx DevTools - Error: ';
+
+type ReduxDevtools = {
+    connect(options?: DevtoolsOptions): DevtoolsConnection;
+};
+
+class DevTools {
+    private remoteDev?: DevtoolsConnection;
+
+    private state: Record<string, unknown>;
+
+    private defaultState: Record<string, unknown>;
+
+    constructor() {
+        this.remoteDev = this.connectViaExtension();
+        this.state = {};
+        this.defaultState = {};
+    }
+
+    connectViaExtension(options?: DevtoolsOptions): DevtoolsConnection | undefined {
+        try {
+            // We don't want to augment the window type in a library code, so we use type assertion instead
+            // eslint-disable-next-line no-underscore-dangle, @typescript-eslint/no-explicit-any
+            const reduxDevtools: ReduxDevtools = (window as any).__REDUX_DEVTOOLS_EXTENSION__;
+
+            if ((options && options.remote) || typeof window === 'undefined' || !reduxDevtools) {
+                return;
+            }
+            // eslint-disable-next-line no-underscore-dangle, @typescript-eslint/no-explicit-any
+            return reduxDevtools.connect(options);
```

Comment:
```suggestion

            return reduxDevtools.connect(options);
```

I think these two rules are unnecessary, I removed here and no problems